tests/infrastructure/taints-and-tolerations.go

```diff
@@ -165,9 +98,145 @@ var _ = DescribeInfra("[rfe_id:4126][crit:medium][vendor:cnv-qe@redhat.com][leve
 			selectedNode, err = virtClient.CoreV1().Nodes().Patch(context.Background(), selectedNode.Name, types.JSONPatchType, patchData, metav1.PatchOptions{})
 			Expect(err).ToNot(HaveOccurred())
 
-			Consistently(terminatedPodsCn, 5*time.Second).ShouldNot(Receive(), "pods should not terminate")
-			stopCn <- true
+			taintedNodeName = nodeName
+		})
+
+		AfterEach(func() {
+			if taintedNodeName == "" {
+				return
+			}
+
+			By("removing the taint from the tainted node")
+			selectedNode, err := virtClient.CoreV1().Nodes().Get(context.Background(), taintedNodeName, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			var taints []k8sv1.Taint
+			for _, taint := range selectedNode.Spec.Taints {
+				if taint.Key != "CriticalAddonsOnly" {
+					taints = append(taints, taint)
+				}
+			}
+
+			patchData, err := patch.GenerateTestReplacePatch("/spec/taints", selectedNode.Spec.Taints, taints)
+			Expect(err).NotTo(HaveOccurred())
+			selectedNode, err = virtClient.CoreV1().Nodes().Patch(context.Background(), selectedNode.Name, types.JSONPatchType, patchData, metav1.PatchOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			// Waiting until all affected deployments have at least 1 ready replica
+			checkedDeployments := map[types.NamespacedName]struct{}{}
+			Eventually(func(g Gomega) {
+				for _, namespacedName := range deploymentsOnNode {
```

Comment:
Maybe unpopular opinion but what about deleting items from this list instead?

Reply:
I thought about it. It would be faster, but more complex code. In this case I prefer simpler to read code.
Because the performance benefit is negligible compared to k8s API calls.